Draw map region overlay

AlexKolpa opened the pull request:
Using the existing 'map squares' dev tool, the world map can now be given an overlay to show all the region boundaries and region ids. This should make it easier to find out the region id of a specific location.

One caveat is that this does not yet work with 'stitched' locations, such as Motherlode Mine.

Example:
![image](https://user-images.githubusercontent.com/3596362/40389137-d3d78fa2-5e11-11e8-9cdb-2f49822d7bda.png)

kulershk commented:
Can you add shadow to text or stroke, because atm you cant read numbers in few places unless u look very closely

AlexKolpa commented:
@kulershk good point, added a translucent background and also moved the label so it's anchored to the grid:

![image](https://user-images.githubusercontent.com/3596362/40452599-b044a27e-5ee2-11e8-8286-eed4af26b59b.png)

deathbeam commented:
Needs rebase

AlexKolpa commented:
Done :+1:

deathbeam commented:
Apparently it needs another rebase :(

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/devtools/WorldMapRegionOverlay.java:
```diff
@@ -0,0 +1,131 @@
+	private final DevToolsPlugin plugin;
+
+	@Inject
+	public WorldMapRegionOverlay(Client client, DevToolsPlugin plugin)
```
Make this constructor private

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/devtools/WorldMapRegionOverlay.java:
```diff
@@ -0,0 +1,131 @@
+		}
+
+		drawRegionOverlay(graphics);
+
```
remove this empty line

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/devtools/WorldMapRegionOverlay.java:
```diff
@@ -0,0 +1,131 @@
+		RenderOverview ro = client.getRenderOverview();
+		Widget map = client.getWidget(WidgetInfo.WORLD_MAP_VIEW);
+		Float pixelsPerTile = ro.getWorldMapZoom();
+		if (map != null)
```
invert this if (e.g if map == null return) also add empty line above

AlexKolpa commented:
Rebased and fixed up the feedback!

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/devtools/WorldMapRegionOverlay.java:
```diff
@@ -0,0 +1,133 @@
+import net.runelite.client.ui.overlay.OverlayPriority;
+
+@Slf4j
+public class WorldMapRegionOverlay extends Overlay
```
Remove public modifier

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/devtools/WorldMapRegionOverlay.java:
```diff
@@ -0,0 +1,133 @@
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+
+@Slf4j
```
I dont see logger used anywhere, remove

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/devtools/WorldMapRegionOverlay.java:
```diff
@@ -0,0 +1,133 @@
+
+		Point worldMapPosition = ro.getWorldMapPosition();
+
+		//Offset in tiles from anchor sides
```
Add space between // and Offset

Matt2408 commented:
I love what you did so far! Is it possible that you are able to add an option to make the grid's 1/4 of the current side. (for even more specific locations)?

AlexKolpa commented:
@deathbeam thanks for the feedback, should be fixed up now!

Kamielvf commented:
I think @Matt2408 is misunderstanding what the plugin shows
